Set config on start

rojasreinold opened the pull request:
Right now there is an error on first run where you are unable to exit. To reproduce delete the whole shoudstone data directory and rerun shroudstone. The app will mostly start fine but the Exit button does nothing.

Error message on cli:
```
INFO     Cache files are from an incompatible version of shroudstone, deleting them.                                                                                         
INFO     Setting renaming config to use replays only - RIP stormgateworld API :(                                                                                             
INFO     Keep this console open - it will show progress information during renaming.                                                                                         
Exception in thread Thread-1 (_tray_icon_thread):
Traceback (most recent call last):
  File "/usr/lib/python3.12/threading.py", line 1073, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.12/threading.py", line 1010, in run
    self._target(*self._args, **self._kwargs)
  File "/home/rrojas/workspace/shroudstone/shroudstone/shroudstone/gui/app.py", line 123, in _tray_icon_thread
    @state.autorename.on_change
     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rrojas/workspace/shroudstone/shroudstone/shroudstone/gui/app.py", line 34, in on_change
    self.trace_add("write", func)
  File "/usr/lib/python3.12/tkinter/__init__.py", line 459, in trace_add
    cbname = self._register(callback)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/tkinter/__init__.py", line 443, in _register
    self._tk.createcommand(cbname, f)
RuntimeError: main thread is not in main loop
````

Fixed by saving the config on first load and choosing the current version as the last run (since there is no last run).

Also updated the readme with build steps.

rojasreinold committed:
Fix unable to close on first run

rojasreinold committed:
Add build instructions to readme

acarapetis commented:
Thanks for this. I can inconsistently reproduce the bug but your change doesn't seem to fix it for me - still get the error sometimes on first startup. Looking into it.

acarapetis committed:
Fix "main thread is not in main loop" errors

These were often seen on first launch, and were caused by the systray
thread attempting to work with Tk variables before the Tk mainloop had
started. I've moved all this custom setup code into a callback that runs
very early inside the mainloop, so hopefully should be smooth sailing
now.

acarapetis commented on shroudstone/renamer.py:
```diff
@@ -76,7 +76,7 @@ def migrate():
         )
     else:
-        last_run_version = version.parse("0.1.0a29")
+        last_run_version = version.parse(__version__)
```
This will mean migrations won't run at all the first time a user launches the app, which isn't great. 
Right now this has no impact because the cache files are no longer used anyway, but could break things in the future.
I'll switch this back.

acarapetis committed:
Revert migrations change

acarapetis committed:
README: Update development instructions

acarapetis commented:
The real problem was that the tray icon thread was sometimes trying to interact with the Tk app before the Tk mainloop had started. I think I've fixed it, though the GUI code is still a mess... at some point it either needs a big cleanup/refactor or maybe just a rewrite in something like Qt that has tray icon support built-in.

Thanks for your contribution! :)
